plugins/modules/nd_site.py

```diff
@@ -0,0 +1,249 @@
+
+
+def validate_not_none(val):
+    return val if val is not None else ""
+
+
+def main():
+
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        apic_password=dict(type="str", no_log=True),
+        apic_username=dict(type="str", default="admin"),
+        apic_login_domain=dict(type="str"),
+        apic_inband_epg=dict(type="str"),
+        site=dict(type="str", aliases=["name"]),
+        url=dict(type="str"),
+        site_type=dict(type="str", choices=["ACI", "DCNM", "ThirdParty", "CloudACI", "DCNMNG"]),
+        security_domains=dict(type="list", elements="str"),
+        latitude=dict(type="float"),
+        longitude=dict(type="float"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["site"]],
+            ["state", "present", ["site", "apic_username", "apic_password", "url", "site_type"]],
+        ],
+    )
+
+    nd = NDModule(module)
+
+    apic_username = nd.params.get("apic_username")
+    apic_password = nd.params.get("apic_password")
+    apic_inband_epg = validate_not_none(nd.params.get("apic_inband_epg"))
+    apic_login_domain = validate_not_none(nd.params.get("apic_login_domain"))
+    site = nd.params.get("site")
+    url = nd.params.get("url")
+    site_type = nd.params.get("site_type")
+    latitude = validate_not_none(nd.params.get("latitude"))
+    longitude = validate_not_none(nd.params.get("longitude"))
+    security_domains = nd.params.get("security_domains")
+    state = nd.params.get("state")
+
+    path = "/api/config/class/v2/sites/"
+
+    # Query for existing object(s)
+    if site:
+        site_info = next((site_dict.get('name') for site_dict in nd.query_obj(path) if site_dict.get('name') == site), None)
+        if site_info:
+            path = "/api/config/dn/v2/sites/{0}".format(site)
+            nd.existing = nd.query_obj(path)
+    else:
+        nd.existing = nd.query_obj(path)
+
+    if state == "query":
+        pass
+    elif state == "absent":
+        nd.previous = nd.existing
+        if nd.existing:
+            if module.check_mode:
+                nd.existing = {}
+            else:
+                rm_path = "/api/config/v2/deletesite/"
+                rm_payload = {
+                    "name": site,
+                    "aci": {
+                        "userName": apic_username,
+                        "password": apic_password,
+                        "loginDomain": apic_login_domain,
+                    }
+                }
+                rm_resp = nd.request(rm_path, method="POST", data=rm_payload)
+                if rm_resp["response"] is not None:
+                    nd.existing = {}
+    elif state == "present":
+        nd.previous = nd.existing
+        add_path = "/api/config/v2"
+        if nd.existing:
+            payload = nd.existing
```

Comment:
Is it not possible to change site information? If the payload is equal to nd.existing what is the use of the PUT when nothing changes?